lib/adagrams.rb

```diff
@@ -0,0 +1,136 @@
+#wave 1
+def draw_letters
+  letters_pool = []
+
+  letter_quantity = {"A": 9, "B": 2, "C": 2, "D": 4, "E": 12, "F": 2, "G": 3, "H": 2, "I": 9, "J": 1, "K": 1, "L": 4, "M": 2, "N": 6, "O": 8, "P": 2, "Q": 1, "R": 6, "S": 4, "T": 2, "U": 4, "V": 2, "W": 2, "X": 1, "Y": 2, "Z": 1}
+
+  letter_quantity.each do |letter, qty|
+    qty.times do
+      letters_pool << letter.to_s
+    end
+  end
+
+  #p letters_pool
+  drawn_letters = letters_pool.sample(10)
+  return drawn_letters
+end
+
+##################################################
+#Wave 2
+def uses_available_letters?(input, letters_in_hand)
+  valid_word = nil
+  input_as_array = input.upcase.split("")
```

Comment:
You can make this a little clearer with `chars`:

```suggestion
  input_as_array = input.upcase.chars
```